### ISSUE TITLE
Customer insights. Match.

### ISSUE THREAD

**Issue by Praxie** [Pliohub/Plio#1295]
![screen shot 2018-08-16 at 5 00 59 pm](https://user-images.githubusercontent.com/2095940/44213273-fa74bc00-a175-11e8-86f8-ba2f467ba8bd.png)

Docs: https://basecamp.com/1896845/projects/11083679/documents/13035477.

**Acceptance criteria:**

- On the LHS there should be all Needs & Wants linked to this section item document.
- On the RHS there should be all Features & Benefits linked to the section item document that is **matched to the current document**. If the matched document has no Features or Benefits or there's no matched document at all, then we should display the following message:
![screen shot 2018-08-16 at 5 04 15 pm](https://user-images.githubusercontent.com/2095940/44213706-014ffe80-a177-11e8-97c7-afbbf00bd447.png)

![screen shot 2018-08-16 at 5 04 15 pm](https://user-images.githubusercontent.com/2095940/44213462-6e16c900-a176-11e8-9fce-4341239c3794.png)

**Comment by Praxie** [comment]
@sives please confirm that if we select a different Value proposition or None in the "Matched to" field, then the whole Match form is to be reset.

**Comment by sives** [comment]
@Praxie I think your question (relating to the image of the Value proposition card above) should be stated as "please confirm that if we select a different Customer segment or None in the "Matched to" field, then the whole Match form is to be reset.

**Comment by sives** [comment]
@Praxie It is not the whole Match form that is reset, just one column in the match form.   So, for the Value proposition card, the Features & benefits column in the Value components subcard remains the same, but the Needs & wants column is reset.    For the Customer segments card, the Needs & wants column stays the same but the Features & benefits column is reset.

**Comment by Praxie** [comment]
@sives yes that's what I meant. Thanks.

**Comment by Praxie** [comment]
The "Unmatch" option should be the same color as other options. It should also have an icon to the LHS - (x) and a separator line underneath.

**Comment by sives** [comment]
The messages that are displayed during the matching process, when there are no items available to match, need to be a bit more helpful, see below.

If the matched document has no Features or Benefits, then we should display the following message:
"No Features or Benefits found. Please add some to the Value proposition"

If there's no matched document at all, then we should display the following message:
"No Value proposition found. Please add one, with some Features and Benefits"

**Comment by Dartv** [comment]
@sives I have a couple of questions. Do we need to display the list of features/benefits in RHS in specific order?
1. Is the benefit/feature displayed to the right of need/want in the same row should be matched to that need/want?
2. What if it's not matched? Do we just leave a blank space?
3. Where to display the rest of unmatched features/benefits?

Or did I get the idea wrong and we just need to display the list of features/benefits

**Comment by sives** [comment]
@Dartv  
1. A benefit/feature is only displayed to the RHS of need/want in the same row once if it has been matched, otherwise nothing is displayed to the RHS (except for the matching list, which is displayed when the user clicks on the double-arrow button)
2. Yes.
3. Unmatched features/benefits aren't displayed anywhere (except of course in the selection list), only matched features/benefits are displayed.
4. Yes, multiple features and/or benefits can be matched to a single need or want in the Customer Segment card.  Multiple needs and/or wants can be matched to a feature or benefit in the Value Proposition card.  (The Match subcard has a very similar design in the Customer Segment and Value Proposition cards, just the order of the two columns is reversed)

**Comment by Dartv** [comment]
@sives If multiple features/benefits can be matched to a single need/want, how the "unmatch" button should behave? Should it unmatch all of the items?

**Comment by sives** [comment]
No, the unmatched function should allow individual items (each carrying their own delete symbol i.e. circle with "x" inside) to be deleted separately

**Comment by Dartv** [comment]
@sives Mike's first screenshot shows RHS item as a simple text (not clickable) and "Unmatch" button inside dropdown menu of "↔️" symbol. UI prototype however is different. Which one of them is correct?

**Comment by Dartv** [comment]
@sives Should benefits/features and needs/wants cross-reference each other? For example, need NE1 is matched to feature FE1 in customer insights subcard. If we go to value components subcard, should we see that the FE1 is matched to NE1?

**Comment by Praxie** [comment]
@sives @Dartv 

> Should benefits/features and needs/wants cross-reference each other? For example, need NE1 is matched to feature FE1 in customer insights subcard. If we go to value components subcard, should we see that FE1 is matched to NE1?

Yes, it's a proper behaviour. if NE1 is matched to FE1 in Customer segments, then in FE1 should be matched to NE1 in Value proposition.

**Comment by Praxie** [comment]
![screen shot 2018-09-12 at 4 14 43 pm](https://user-images.githubusercontent.com/2095940/45427246-0a18ef00-b6a7-11e8-910e-d3752ec4927f.png)

**Comment by Praxie** [comment]
@sives what to you think about this design?

**Comment by Praxie** [comment]
![screen shot 2018-09-12 at 4 18 03 pm](https://user-images.githubusercontent.com/2095940/45427394-7398fd80-b6a7-11e8-9083-68448e57b7f5.png)

**Comment by sives** [comment]
@Praxie Looks good to me. @Dartv Yes, you are correct, benefits/features and needs/wants should exactly cross-reference each other.

**Comment by Dartv** [comment]
https://github.com/Pliohub/Plio/pull/1391
https://github.com/Pliohub/Plio/pull/1390
https://github.com/Pliohub/Plio/pull/1386
https://github.com/Pliohub/Plio/pull/1385
https://github.com/Pliohub/Plio/pull/1373

**Comment by Dartv** [comment]
via Hubstaff
User: Dmytro Artamonov

Project: Plio Connected - https://app.hubstaff.com/projects/329669
Date Range: 09/07/18 - 09/19/18
Work session total: 28:29:31

Grand total: 28:29:31
